### ISSUE TITLE
The server remained unresponsive for a long time when I try to train your model.

### ISSUE THREAD

**Issue by longmalongma** [hkchengrex/Mask-Propagation#4]
When I ran this line of code on our server, the server did not respond for a long time. Do you know why?

UDA_VISIBLE_DEVICES=0 OMP_NUM_THREADS=4 python -m torch.distributed.launch --master_port 9842 --nproc_per_node=1 train.py --id retrain_s0 --stage 0 --batch_size 4

**Comment by hkchengrex** [comment]
Did you miss the "C" at the front?

**Comment by hkchengrex** [comment]
but that's probably not the reason... Which line is it stuck at? You can use a tool to check: https://github.com/benfred/py-spy

**Comment by longmalongma** [comment]
> but that's probably not the reason... Which line is it stuck at? You can use a tool to check: https://github.com/benfred/py-spy

Thanks for your reply. I will try it.

**Comment by hkchengrex** [comment]
@longmalongma It seems that this problem has not been fixed. How about you try some simple debugging (like printing 1, 2, 3, 4) at different lines of `train.py` and see which line does the program stuck at?

**Comment by longmalongma** [comment]
> @longmalongma It seems that this problem has not been fixed. How about you try some simple debugging (like printing 1, 2, 3, 4) at different lines of `train.py` and see which line does the program stuck at?



> @longmalongma It seems that this problem has not been fixed. How about you try some simple debugging (like printing 1, 2, 3, 4) at different lines of `train.py` and see which line does the program stuck at?

OK, now I will try it according to your suggestion.

**Comment by longmalongma** [comment]
> > @longmalongma It seems that this problem has not been fixed. How about you try some simple debugging (like printing 1, 2, 3, 4) at different lines of `train.py` and see which line does the program stuck at?
> 
> > @longmalongma It seems that this problem has not been fixed. How about you try some simple debugging (like printing 1, 2, 3, 4) at different lines of `train.py` and see which line does the program stuck at?
> 
> OK, now I will try it according to your suggestion.

![image](https://user-images.githubusercontent.com/40291722/113829979-b74d3880-97b8-11eb-8fec-3b7861198029.png)
I can't print anything, it stays in this interface.

**Comment by longmalongma** [comment]
> @longmalongma It seems that this problem has not been fixed. How about you try some simple debugging (like printing 1, 2, 3, 4) at different lines of `train.py` and see which line does the program stuck at?

@hkchengrex Is there a problem with the data set?
![image](https://user-images.githubusercontent.com/40291722/113830846-96391780-97b9-11eb-9684-933e7a3c2885.png)

**Comment by hkchengrex** [comment]
If you cannot even finish importing it's not because of the dataset...
Can you add a few more print statements to see where does it fail?

**Comment by longmalongma** [comment]
> If you cannot even finish importing it's not because of the dataset...
> Can you add a few more print statements to see where does it fail?

Sorry, none of the rows can be printed, I think it's probably a problem with the data set, because the difference between me and others is the download of the data set.

**Comment by longmalongma** [comment]
> > If you cannot even finish importing it's not because of the dataset...
> > Can you add a few more print statements to see where does it fail?
> 
> Sorry, none of the rows can be printed, I think it's probably a problem with the data set, because the difference between me and others is the download of the data set.



> > If you cannot even finish importing it's not because of the dataset...
> > Can you add a few more print statements to see where does it fail?
> 
> Sorry, none of the rows can be printed, I think it's probably a problem with the data set, because the difference between me and others is the download of the data set.
I found a problem. It got stuck when introducing Torch, but my environment can run STM. Do I need to update the version of Torch?
![image](https://user-images.githubusercontent.com/40291722/113833695-8c64e380-97bc-11eb-841f-cfd954539749.png)

**Comment by hkchengrex** [comment]
Maybe you can uninstall and re-install pytorch?

**Comment by longmalongma** [comment]
> Maybe you can uninstall and re-install pytorch?
ok,my current environment is like this:
torchaudio                0.8.1                      py38    pytorch

**Comment by longmalongma** [comment]
> Maybe you can uninstall and re-install pytorch?

I uninstalled and reinstalled PyTorch. The improved version still doesn't work, and the runtime is still stuck in the Import Torch line.

**Comment by hkchengrex** [comment]
What if you open up python, and only import torch?

**Comment by longmalongma** [comment]
> What if you open up python, and only import torch?
I can import torch when I use the command line to enter Python, but I'm stuck in the import torch line when I run  codes. Why?
![image](https://user-images.githubusercontent.com/40291722/113971749-f5576480-986b-11eb-9d33-e06e8995ab18.png)

**Comment by hkchengrex** [comment]
What if you import torch first, before everything else? i.e., move `import torch` to the first line.

**Comment by longmalongma** [comment]
> move `import torch` to the first line

When I moved the Import Torch to the first line, the Import Torch succeeded. Thank you for your help.

**Comment by longmalongma** [comment]
> What if you import torch first, before everything else? i.e., move `import torch` to the first line.

Why is Import Torch correct on the first line, but incorrect on any other line?Do you know why?I'm being hard to understand.

**Comment by hkchengrex** [comment]
I have no idea. Maybe you can ask the PyTorch guys: https://github.com/pytorch/pytorch/issues

**Comment by longmalongma** [comment]
> I have no idea. Maybe you can ask the PyTorch guys: https://github.com/pytorch/pytorch/issues

ok, thanks!
